Objects/stringlib/codecs.h

```diff
@@ -220,10 +220,10 @@ STRINGLIB(utf8_decode)(const char **inptr, const char *end,
             }
             ch = (ch << 18) + (ch2 << 12) + (ch3 << 6) + ch4 -
                  ((0xF0 << 18) + (0x80 << 12) + (0x80 << 6) + 0x80);
-            assert ((ch > 0xFFFF) && (ch <= 0x10FFFF));
+            assert ((ch > 0xFFFF) && (ch <= MAX_UNICODE));
```

Comment:
Note that the current UTF-8 codec would not work for MAX_UNICODE > 0x001FFFFF.

Reply:
That's the UTF-8 decoder: it has to ensure that it doesn't produce a code point not supported by Python. (ch <= MAX_UNICODE) is the correct test here.